File: Data Structures & Algorithms/anagram-groups/submission-2.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>> mmap;
        for(auto str:strs)
        {
            string key=str;
            sort(key.begin(),key.end());
            mmap[key].push_back(str);
        }
        vector<vector<string>> result;
        for(auto pair:mmap)
        {
            result.push_back(pair.second);
        }
        return result;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-2.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        mmap=defaultdict(int)
        if len(s) !=len(t):
            return False
        for i in range(len(s)):
            mmap[s[i]]+=1
            mmap[t[i]]-=1
        for i in range(len(s)):
            if mmap[s[i]]!=0:
                return False
            
        return True